readthedocs/rtd_tests/tests/test_unresolver.py

```diff
@@ -205,6 +205,40 @@ def test_unresolver_external_version(self):
         self.assertEqual(parts.version, version)
         self.assertEqual(parts.filename, "/index.html")
 
+    def test_external_version_single_version_project(self):
+        self.pip.single_version = True
+        self.pip.save()
+        version = get(
+            Version,
+            project=self.pip,
+            type=EXTERNAL,
+            slug="10",
+            active=True,
+        )
+        parts = unresolve("https://pip--10.dev.readthedocs.build/")
+        self.assertEqual(parts.parent_project, self.pip)
+        self.assertEqual(parts.project, self.pip)
```

Comment:
Why `parent_project` is `self.pip` here?

This project does not have any parent, so I'd say it should be `None`.

Reply:
parent project is the project that owns the domain

https://github.com/readthedocs/readthedocs.org/blob/8c37d1458e5cb43ffd9518b69e8cc2dd58faef84/readthedocs/core/unresolver.py#L20-L22